plugins/cpp_metrics/test/sources/parser/lackofcohesion.cpp

```diff
@@ -0,0 +1,223 @@
+////////////////
```

Comment:
The tests fail in the CI, the PR cannot be merged this way.

Reply:
I cannot figure out why the tests fail though. It's not about the results, it's about the existence of the results at all. It seems as if the types I am looking for are simply not present in the database when the tests run.
I cannot use the test case for McCabe as a reference because that test actually only uses results parsed by the main cpp plugin. In my tests, I require records from the database parsed by the metrics plugin itself.
Is it possible that the tests are not properly configured and the test files that I have included are not even parsed by the metrics parser, only the cpp parser? Is there a way I can check this?

Reply:
@dbukki In that case you should verify the content of the database and see why is there a mismatch?

Reply:
In the tests only the `compile_commands.json` was added to the input path, the source folder itself not. Fixed in [4f43dd7](https://github.com/Ericsson/CodeCompass/pull/688/commits/4f43dd7ee550b552e1e13bf910708082a5ae0f31).